### PR TITLE
Create Button and Header components

### PR DESCRIPTION
**Pull request by hellomkreyes**
## Description
Created new components for **Button** and **Heading**. Rendering API response when `Generate Caption` is clicked and styling layout to be 2 column grid.

### New components
1. `src/common/components/Button/Button.js`: Button
1. `src/common/components/Button/index.js`: export button
1. `src/common/components/Heading/Heading.js`: Heading
1. `src/common/components/Heading/index.js`: export heading
1. `src/common/components/index.js`: export new components

### Changes
1. `src/common/components/ImageUploader/index.js`: import ImageUploader
1. `src/common/components/ImageUploader/ImageUploader.js`: use Button component + add image file types for input and image preview
1. `src/core/App.js`: use heading component + save API response in state + output API response
1. `src/core/App.module.scss`: grid styling

### CODE REVIEW

File: src/common/components/Button/Button.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+export const Button = ({ children }, props) => {
```
Comment (mshelley-twg) [inline src/common/components/Button/Button.js]: If you want to get the rest of the props, you can do:

```({ children, ...rest }) => { ... }```

File: src/common/components/Heading/Heading.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+export const Heading = (props) => {
+  const level = props.level || 2
```
Comment (mshelley-twg) [inline src/common/components/Heading/Heading.js]: In React, we can use `defaultProps` instead! Whenever a prop is not provided, we get the default.

```
Heading.defaultProps = {
  level: 2
}
```

File: src/core/App.module.scss [hunk 3]
```diff
@@ -1,22 +1,28 @@
...
+  &-link {
+    color: #09d3ac;
+  }
+
+  &-grid {
+    max-width: 1300px;
+    margin: 0 auto;
+    display: flex;
+  }
+}
```
Comment (mshelley-twg) [inline src/core/App.module.scss]: Please add line breaks to the end of your files.

It seems unnecessary, but editors and GitHub prefer it. To get technical, the definition of a line of text requires a line break.

File: src/common/components/ImageUploader/ImageUploader.js [hunk 4]
```diff
@@ -58,15 +59,24 @@ class ImageUploader extends Component {
...
     return (
       <>
         <form onSubmit={this.handleSubmit}>
-          <input type='file' onChange={this.handleImageChange} />
-          <button type='submit' onClick={this.handleSubmit}>Generate Caption!</button>
+          <label htmlFor='imageFile'>File Upload</label>
+          <input type='file'
+            id='imageFile'
+            name='imageFile'
+            accept='image/png, image/jpeg'
```
Comment (mshelley-twg) [inline src/common/components/ImageUploader/ImageUploader.js]: Nice! Happy to see you thinking of accessibility and forms :)

File: src/core/App.module.scss (hunk 3 above)
Comment (hellomkreyes) [reply src/core/App.module.scss]: Thanks @mshelley-twg for the context! I never knew the reason why end of file line breaks were necessary.

**Pull request merged**